worker/baggageclaim/api/volume_server.go

```diff
@@ -588,29 +592,47 @@ func (vs *VolumeServer) StreamP2pOut(w http.ResponseWriter, req *http.Request) {
 		return
 	}
 
-	err := vs.volumeRepo.StreamP2pOut(ctx, handle, subPath, encoding, streamInURL)
+	rawUrl, err := url.Parse(streamInURL)
 	if err != nil {
-		if err == volume.ErrVolumeDoesNotExist {
-			hLog.Info("volume-not-found")
-			RespondWithError(w, ErrStreamOutNotFound, http.StatusNotFound)
-			return
-		}
+		hLog.Info("bad-param-streamInURL")
+		RespondWithError(w, ErrStreamP2pOutFailed, http.StatusBadRequest)
+	}
+	streamInURL = rawUrl.String()
 
-		if err == volume.ErrUnsupportedStreamEncoding {
-			hLog.Info("unsupported-stream-encoding")
-			RespondWithError(w, ErrStreamP2pOutFailed, http.StatusBadRequest)
-			return
-		}
+	w.Header().Set("Content-Type", "plain/text")
+	w.Header().Set("X-Content-Type-Options", "nosniff")
 
-		if os.IsNotExist(err) {
-			hLog.Info("source-path-not-found")
-			RespondWithError(w, ErrStreamOutNotFound, http.StatusNotFound)
+	doneChan := make(chan error, 1)
+	go func(doneChan chan<- error) {
+		err := vs.volumeRepo.StreamP2pOut(ctx, handle, subPath, encoding, streamInURL)
+		if err != nil {
+			hLog.Error("failed-to-stream-out", err)
+			doneChan <- fmt.Errorf("%s: %w", ErrStreamP2pOutFailed, err)
+		} else {
+			close(doneChan)
+		}
+	}(doneChan)
+
+	// Send a white space to client as an indicator of in-progress every 10 seconds.
+	// After source worker finishes sending volume ot dest worker, send client "ok"
+	// for success or an error message.
+	tick := time.NewTicker(10 * time.Second)
```

Comment:
if the timeout is 3min, is ticking in 10 sec necessary here? I am assuming every tick will refrest the 3min timeout timer.

Reply:
3 min is configurable. A p2p streaming may take uncertain time, maybe a few seconds, maybe 10 minutes or even longer. So I believe a periodical progress helps maintain the HTTP transaction stable. You may argue is 10 sec interval is short, that is fine, I don't consider much difference for 10 sec, 30 sec or 1 min, which should not be big overhead.

Reply:
Could you make it 30sec(same as workers beacon to ATC)?